shell/platform/linux/fl_renderer_x11.cc

```diff
@@ -27,6 +27,14 @@ static GdkVisual* fl_renderer_x11_get_visual(FlRenderer* renderer,
   return gdk_x11_screen_lookup_visual(GDK_X11_SCREEN(screen), visual_id);
 }
 
+// Implements FlRenderer::set_window.
+static void fl_renderer_x11_set_window(FlRenderer* renderer,
+                                       GdkWindow* window) {
+  FlRendererX11* self = FL_RENDERER_X11(renderer);
+  g_return_if_fail(GDK_IS_X11_WINDOW(window));
+  self->window = GDK_X11_WINDOW(g_object_ref(window));
```

Comment:
Might be helpful to put an assertion above this that `self->window == NULL` in case a change in the future causes a leak.